src/nxt_unit.h

```diff
@@ -137,14 +137,9 @@ struct nxt_unit_callbacks_s {
     /* Gracefully quit the application. Optional. */
     void     (*quit)(nxt_unit_ctx_t *);
 
-    /* Send data and control to process pid using port id. Optional. */
-    ssize_t  (*port_send)(nxt_unit_ctx_t *, nxt_unit_port_id_t *port_id,
-                 const void *buf, size_t buf_size,
-                 const void *oob, size_t oob_size);
-
-    /* Receive data on port id. Optional. */
-    ssize_t  (*port_recv)(nxt_unit_ctx_t *, nxt_unit_port_id_t *port_id,
-                 void *buf, size_t buf_size, void *oob, size_t oob_size);
+    /* Lookup the port pair from the port id. Optional. */
+    void     (*lookup_port_pair)(nxt_unit_ctx_t *, nxt_unit_port_id_t *port_id,
+                                int *in_fd, int *out_fd);
```

Comment:
Maybe the prototype below is better:

```c
void (*lookup_port_pair)(nxt_unit_ctx_t *, nxt_unit_port_id_t *port_id, int pair[2]);
```

Let me know what you think.
@VBart @mar0x

Reply:
I think we should avoid blocking in ListenAndServe() function.  In spite of the fact cgo function calls slower than pure Go function, excessive cgo calls looks better than locking the thread with a Unit wait message loop.
For code reuse, would be nice to investigate if it is possible to poll the fd in Go without reading, and then read and process message in C - this way we can reuse nxt_readmsg().

Reply:
Ok, I'm closing this as it requires a better solution.
Thanks